bundler/spec/dependabot/bundler/native_helpers_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "dependabot/bundler/native_helpers"
+
+RSpec.describe Dependabot::Bundler::NativeHelpers do
+  subject { described_class }
+
+  describe ".run_bundler_subprocess" do
+    context "with a timeout provided" do
+      it "terminates the spawned process when the timeout is exceeded" do
+        allow(Dependabot::SharedHelpers).to receive(:run_helper_subprocess)
+
+        subject.run_bundler_subprocess(
+          function: "noop",
+          args: [],
+          bundler_version: "2.0.0",
+          options: {
+            timeout_per_operation_seconds: 1
```

Comment:
I think historically we have used the experiments hash as effectively a feature flag, so I was surprised to see this passed as an integer value.

I think this approach is fine though, but in the spirit of checking your inputs, should we place a hard cap on the value passed in the event the calling code makes a magnitude error and passes in 'milliseconds' instead of 'seconds' or something like that?

Reply:
> should we place a hard cap on the value passed in the event the calling code makes a magnitude error and passes in 'milliseconds' instead of 'seconds' or something like that?

Good point. How does an upper bound of 30 minutes and a lower bound of 1 minute sound?